client/client.go

```diff
@@ -146,29 +162,34 @@ func (client *Client) reconnect(err error) error {
 	// closed by Gearmand, the client should close the conection
 	// and reconnect to job server.
 	client.Close()
-	client.conn, err = net.Dial(client.net, client.addr)
+
+	conn, err := net.Dial(client.net, client.addr)
 
 	if err != nil {
 		client.err(err)
 		return err
 	}
 
-	client.rw = bufio.NewReadWriter(bufio.NewReader(client.conn),
-		bufio.NewWriter(client.conn))
+	swapped := atomic.CompareAndSwapPointer(
+		(*unsafe.Pointer)(unsafe.Pointer(&client.conn)),
+		unsafe.Pointer(nil),
+		unsafe.Pointer(&connection{conn}))
+
+	if !swapped {
+		return fmt.Errorf("client: unable to connect")
```

Comment:
Why does a failure to swap mean unable to connect? Is this using CompareAndSwapPointer as an equals operator? Why not just equals?

Reply:
Decided not to change?